File: Petstaurant/Models/Store.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Petstaurant.Models
{
    public class Store
    {
        public int Id { get; set; }
        [StringLength(20, MinimumLength = 2, ErrorMessage = "You are allowed to use only 2-20 characters")]
        [RegularExpression(@"[a-zA-Z]+(?:[ '-][a-zA-Z]+)*", ErrorMessage =
           "Please choose a valid city name")]
        public string City { get; set; }
        [RegularExpression(@"^[#.0-9a-zA-Z\s,-]+$", ErrorMessage =
           "Israeli address format is required (Example: Israel Galili 5)")]
        [StringLength(30, MinimumLength = 4, ErrorMessage = "You are allowed to use only 4-30 characters")]
        public string Address { get; set; }
        [Required]
        [RegularExpression(@"^\d{7}$", ErrorMessage =
           "Israeli postal code format is required")]
        [DisplayName("Postal Code")]
        [StringLength(7, MinimumLength = 7, ErrorMessage = "You are allowed to use only 7 digits")]
        [DataType(DataType.PostalCode)]
        public string PostalCode { get; set; }
        [DisplayName("Dishes")]
        public List<Dish> Dish { get; set; }
        public List<Order> Orders { get; set; }

    }
}

File: Petstaurant/wwwroot/js/store.js
$(function () {
    $('#storeValidation').on("submit", function (e) {
        var ad = $('#address').val();
        var adEr = $('#addressError');
        var c = false;
        var a = ad.split(" ");
        if (a.length < 2 || a.length>4) {
            c = true;
        }
        for (var i = 0; i < a.length; i++) {
            if (a[i].match(/.*([a-zA-Z].*[0-9]|[0-9].*[a-zA-Z]).*/)) {
                c = true;
                    break;
            }
        }
        if (parseInt(a[0])) {
            c = true;
        }

        if (c || ad.startsWith(" ") || ad.endsWith(" ") || ad.includes(",") || parseInt(ad) || ad.match(/\s/g).length > a.length) {
        e.preventDefault();
            adEr.text('Israeli address format is required (Example: Israel Galili 5)');
        }
        c = false;

        var po = $('#postalCode').val();
        var poE = $('#postalCodeError');
        if (po.startsWith("0")) {
            e.preventDefault();
            poE.text('Israeli postal code format is required');
        }
    });
});